akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala

```diff
@@ -637,30 +637,15 @@ import scala.util.control.NonFatal
     }
   }
 
-  override def toString: String = {
-    val builder = StringBuilder.newBuilder
-    builder.append("GraphInterpreterShell(\n  logics: [\n")
-    val logicsToPrint = if (isInitialized) interpreter.logics else logics
-    logicsToPrint.foreach { logic ⇒
-      builder.append("    ")
-        .append(logic.originalStage.getOrElse(logic).toString)
-        .append(" attrs: [")
-        .append(logic.attributes.attributeList.mkString(", "))
-        .append("],\n")
-    }
-    builder.setLength(builder.length - 2)
-    if (isInitialized) {
-      builder.append("\n  ],\n  connections: [\n")
-      interpreter.connections.foreach { connection ⇒
-        builder
-          .append("    ")
-          .append(if (connection == null) "null" else connection.toString)
-          .append(",\n")
-      }
-      builder.setLength(builder.length - 2)
-    }
-    builder.append("\n  ]\n)")
-    builder.toString()
+  def toSnapshot: InterpreterSnapshot = {
```

Comment:
should we still override `toString` to print this snapshot?

Reply:
I don't think toString will ever be actually called